### PR TITLE
Standard js

### PR DESCRIPTION
**Pull request by Schultzer**

### CODE REVIEW

File: e2e/mocks.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-const electron = require('electron');
+const path = require('path')
+const electron = require('electron')
 electron.dialog.showOpenDialog = (opts, cb) => {
-  cb([__dirname + '/support/readable.pdf']);
-};
+  cb([path.join(__dirname, '/support/readable.pdf')])
```
Comment (danschultzer) [inline e2e/mocks.js]: This doesn't really make sense unless you separate each level so no slash is used.

File: e2e/app.e2e.js [hunk 2]
```diff
@@ -1,136 +1,134 @@
...
+        })
+    })
+
+    describe('when copying', function () {
+      it('should copy CSV to clipboard', function () {
         return this.app.client.click('.table .body .row')
-          .execute(function() {
-            document.dispatchEvent(new Event('copy'));
+          .execute(function () {
+            document.dispatchEvent(new window.Event('copy'))
```
Comment (danschultzer) [inline e2e/app.e2e.js:48]: Why `window.Event` and not `Event`? It's a std class after all.

File: src/background.js [hunk 3]
```diff
@@ -3,168 +3,169 @@
...
+  opts.titleBarStyle = opts.titleBarStyle || 'hidden'
 
-  var win = createWindow(windowName, opts);
+  var win = createWindow(windowName, opts)
 
-  blur(win, 'report-' + windowName);
+  blur(win, 'report-' + windowName)
 
-  win.loadURL('file://' + __dirname + file);
+  win.loadURL(path.join('file://', __dirname, file))
```
Comment (danschultzer) [inline src/background.js]: Why this? It should be URL module handling this, it's not a system path, but a URL!

Comment (danschultzer) [reply src/background.js]: But I don't think it's right to use any module to concat this stuff.

File: src/edit.js [hunk 4]
```diff
@@ -1,158 +1,159 @@
...
     case /^image\/.*/.test(mimetype):
-      if (!loaded_img) {
-        var img = new Image();
-        img.onload = function() {
-          loaded_img = img;
-          renderImage(loaded_img, canvas, context, maxWidth, maxHeight);
-        };
-        img.src = path;
+      if (!loadedImg) {
+        var img = new window.Image()
```
Comment (danschultzer) [inline src/edit.js:75]: Why `window.Image`?

File: src/file_list/file_list.js [hunk 5]
```diff
@@ -1,53 +1,53 @@
...
 
-  getFileIndexForIndex(index) {
-    return this.files.findIndex(function(f) { return f.index === index; });
+  getFileIndexForIndex (index) {
+    return this.files.findIndex(function (f) { return f.index === index })
   }
 
-  createSmoothPercentProgressionInterval(index) {
+  createSmoothPercentProgressionInterval (index) {
     var interval = new Util.Interval(() => {
```
Comment (danschultzer) [inline src/file_list/file_list.js:35]: Here's is the usage of ECMA6 function definition. It should be kept consistent throughout, and it might be good to just switch all the code over to ECMA6 (?) This should be done in another PR though.

File: src/file_list/file_list.js [hunk 6]
```diff
@@ -63,133 +63,134 @@ export class FileList {
...
+      var file = this.queued()[0]
+      var index = file.index
+      var name = file.file.name
+      var path = file.file.path
+
+      this.updateFile(index, {
+        processing: true
+      })
+
+      tmp.tmpNameSync({ postfix: name.substr(name.lastIndexOf('.')) })
```
Comment (danschultzer) [inline src/file_list/file_list.js]: Why is this here? It's not used for anything, so remove it.

File: src/file_list/file_list.js [hunk 7]
```diff
@@ -63,133 +63,134 @@ export class FileList {
...
         files[i].result && files[i].result.parsed ? (files[i].result.updated || {}).amount || files[i].result.parsed.amount : '',
         files[i].result && files[i].result.parsed ? (files[i].result.updated || {}).date || files[i].result.parsed.date : '',
         files[i].file.path
-      ];
-      values = values.map(function(value) {
-        if (value && value.length && (value.indexOf('"') > -1 || value.indexOf('\t') > -1))
-          return '"' + value.replace('"', '\\"') + '"';
+      ]
+      values = values.map(function (value) {
+        if (value && value.length && (value.indexOf('"') > -1 || value.indexOf('\t') > -1)) {
```
Comment (danschultzer) [inline src/file_list/file_list.js:163]: Should probably have a replace for `\t` as well if it's part of the logic.

File: src/file_list/select.js [hunk 8]
```diff
@@ -1,163 +1,165 @@
...
-  indexOf(item) {
-      var children = this.items(),
-          i = 0;
-      for (; i < children.length; i++) {
-          if (children[i] === item) {
-              return i;
-          }
+  indexOf (item) {
+    var children = this.items()
+    var i = 0
```
Comment (danschultzer) [inline src/file_list/select.js]: This could just be put in the for loop below?

File: src/file_list/util.js [hunk 9]
```diff
@@ -1,92 +1,91 @@
...
+    filelist = filelist || []
     files.forEach((file) => {
-        if (this.isDir(dir + '/' + file)) {
-            filelist = this.walkSync(dir + '/' + file, filelist);
-        } else {
-            filelist.push(dir + '/' + file);
-        }
-    });
-    return filelist;
+      if (this.isDir(dir + '/' + file)) {
```
Comment (danschultzer) [inline src/file_list/util.js]: If you want to use `path.join`, then it should be consistent and used anywhere with path separators.

**Review by danschultzer — CHANGES_REQUESTED** [review]

File: e2e/app.e2e.js (hunk 2 above)
Comment (Schultzer) [reply e2e/app.e2e.js:48]: Issue [562](https://github.com/feross/standard/issues/562) and [619](https://github.com/feross/standard/pull/619)
we can go either way `window.` or a comment `/* global */`

File: src/edit.js (hunk 4 above)
Comment (Schultzer) [reply src/edit.js:75]: Issue [562](https://github.com/feross/standard/issues/562) and [619](https://github.com/feross/standard/pull/619)
we can go either way `window.` or a comment `/* global */`

File: src/file_list/select.js [hunk 10]
```diff
@@ -12,10 +12,8 @@ export default class Select {
   }
 
   indexOf (item) {
-    var children = this.items()
-    var i = 0
-    for (; i < children.length; i++) {
-      if (children[i] === item) {
+    for (var i = 0; i < this.items().length; i++) {
+      if (this.items()[i] === item) {
```
Comment (danschultzer) [inline src/file_list/select.js]: This is not good practice, keep children var to not call this every single iteration. What happens if items() suddenly changes? Can cause infinite loop.

File: src/file_list/file_list.js [hunk 11]
```diff
@@ -162,7 +162,7 @@ export class FileList {
       ]
       values = values.map(function (value) {
         if (value && value.length && (value.indexOf('"') > -1 || value.indexOf('\t') > -1)) {
-          return '"' + value.replace('"', '\\"') + '"'
+          return '"' + value.replace('"', '\\"', '\t') + '"'
```
Comment (danschultzer) [inline src/file_list/file_list.js]: Remove this commit, I overlooked that this just adds quotes around so no need for it. Also the replace is all wrong, but doesn't matter.

Comment (danschultzer) [reply src/file_list/file_list.js]: This needs to be fixed still, not sure why it's outdated

File: src/file_list/file_list.js [hunk 12]
```diff
@@ -63,133 +62,132 @@ export class FileList {
...
         files[i].result && files[i].result.parsed ? (files[i].result.updated || {}).date || files[i].result.parsed.date : '',
         files[i].file.path
-      ];
-      values = values.map(function(value) {
-        if (value && value.length && (value.indexOf('"') > -1 || value.indexOf('\t') > -1))
-          return '"' + value.replace('"', '\\"') + '"';
+      ]
+      values = values.map(function (value) {
+        if (value && value.length && (value.indexOf('"') > -1 || value.indexOf('\t') > -1)) {
+          return '"' + value.replace('"', '\\"', '\t') + '"'
```
Comment (danschultzer) [inline src/file_list/file_list.js]: Still not changed?

File: src/edit.js (hunk 4 above)
Comment (danschultzer) [reply src/edit.js:75]: Let's just use `window.`

**Review by danschultzer — APPROVED** [review]

**Comment by danschultzer** [comment]
Just needs to be manually tested to make sure everything still works (our specs doesn't cover all), and then this can be merged in.
